markokajzer/discord-soundbot
# Upgrade to discord.js@12.0.0

## What has been done

+ Upgrade to [`discord.js` v12](https://github.com/discordjs/discord.js/releases/tag/12.0.0)
+ Use `@discordjs/opus` to avoid deprecated and abandoned `node-opus` package
+ Upgraded to Node v12
+ Add `.node-version` file

## How to test

+ Confirm that every piece of functionality still works as expected
  + Check all commands
  + Check entrance / exit sounds
  + Check all config options

## Notes

+ ~~Node v12 (newest) was not possible due to a transitive dependency of `node-opus` to a package called `ref` which is incompatible with Node v12~~
  + ~~This is definitely a reason to ditch `node-opus` entirely in favor of `opusscript`~~
  + ~~`opusscript` is said to be not very performant in the [discord.js docs](https://github.com/discordjs/discord.js#audio-engines) but how big is the difference? Is it really worth this whole hassle? I couldn't find any mentions of this outside of that README and it has been there since forever. Maybe the issues have been resolved?~~
  + ~~Could also try `ref-napi` according to [this comment](https://github.com/Rantanen/node-opus/issues/93#issuecomment-570787267)~~
